cvp_mesh_planner/src/cvp_mesh_planner.cpp

```diff
@@ -105,8 +105,7 @@ uint32_t CVPMeshPlanner::makePlan(const geometry_msgs::msg::PoseStamped& start,
 
     geometry_msgs::msg::PoseStamped pose;
     pose.header = header;
-    pose.pose = mesh_map::calculatePoseFromPosition(vec, goal_vec, face_normals[fH], dir_length);
-    cost += dir_length;
+    pose.pose = goal.pose;
```

Comment:
Nice. I agree that the last element of the resulting path should probably be exactly the goal pose.
The previous implementation had the additional effect of aligning roll/pitch of the goal pose to the mesh's curvature. That might be nice, but I don't think it's that important.
I think it's essential to keep the yaw component of the original goal pose :+1: 

Thinking about goal orientations brings up a general problem the cvp planner has: It doesn't consider orientations. With your fix, we might end up of with really weird paths in which the robot's orientation needs to flip by 180 degree at the end. But without your fix, the robot's orientation will be just plain wrong in the end, so I think it's a good fix, as it will work well for some cases (e.g. holonomic robots)